notification/rule/teams.go

```diff
@@ -0,0 +1,127 @@
+package rule
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/notification/endpoint"
+	"github.com/influxdata/influxdb/v2/notification/flux"
+)
+
+// Teams is the notification rule config of Microsoft Teams.
+type Teams struct {
+	Base
+	Title           string `json:"title"`
+	MessageTemplate string `json:"messageTemplate"`
+	Summary         string `json:"summary"`
+}
+
+// GenerateFlux generates a flux script for the teams notification rule.
+func (s *Teams) GenerateFlux(e influxdb.NotificationEndpoint) (string, error) {
+	teamsEndpoint, ok := e.(*endpoint.Teams)
+	if !ok {
+		return "", fmt.Errorf("endpoint provided is a %s, not a Teams endpoint", e.Type())
+	}
+	p, err := s.GenerateFluxAST(teamsEndpoint)
+	if err != nil {
+		return "", err
+	}
+	return ast.Format(p), nil
+}
+
+// GenerateFluxAST generates a flux AST for the teams notification rule.
+func (s *Teams) GenerateFluxAST(e *endpoint.Teams) (*ast.Package, error) {
+	f := flux.File(
+		s.Name,
+		flux.Imports("influxdata/influxdb/monitor", "contrib/sranka/teams", "influxdata/influxdb/secrets", "experimental"),
```

Comment:
not being familiar with how these imports work, is `contrib/sranka/teams` outdated since https://github.com/influxdata/flux/pull/2866 got merged?

Reply:
https://github.com/influxdata/flux/tree/master/stdlib/contrib/sranka/teams is still the same, there are tests that make the this package stable. I also tested that the notification works manually using a new UI.